PyPi Intergration

yeslayla opened the pull request:
- **Renamed & Reconfigured Package**
The `setup.py` file was updated to use the package name `semver-by-branch`. The metadata was also updated to work with PyPi.
- **PIPELINE: Split Release Stage in Two**
The `Release` stage was split into the `Release Packages` stage and `Push Docker Image` stage. The `Release Packages Stage` pushes packages to GitHub & PyPi while the `Push Docker Image` stage pushes the `latest` tag to DockerHub.
- **DOCUMENTATION: Updates `README.md`**
The `README.md` file was updated to reflect installation using pip from PyPi.

yeslayla committed:
Update package name

Update to work with PyPI

Fix pipeline

Update pipeline

Update pipeline

Update pipeline

josephbmanley commented on setup.py:
```diff
@@ -34,16 +34,11 @@ def find_version(*file_paths):
-    # Versions should comply with PEP440.  For a discussion on single-sourcing
-    # the version across setup.py and the project code, see
-    # https://packaging.python.org/en/latest/single_source_version.html
+    name='semver-by-branch',
```
If we need to change the package name, that only has to be updated here.

dejonghe commented on Jenkinsfile:
```diff
@@ -81,20 +75,52 @@ pipeline {
+
+        // Upload package to PyPi
+        withCredentials([usernamePassword(credentialsId: env.PYPI_CREDENTIALS, usernameVariable: 'PYPI_USERNAME', passwordVariable: 'PYPI_PASSWORD')]) {
+          sh("pip install twine")
```
can this be done in a docker agent? I'd prefer to not have packages installed on our jenkins.

josephbmanley replied:
Within Docker image

dejonghe commented on Jenkinsfile:
```diff
@@ -39,11 +40,6 @@ pipeline {
         sh "docker build ${getDockerBuildFlags()} -t rightbrainnetworks/auto-semver:${env.VERSION} ."
-
-
-        sh "python setup.py sdist"
```
If you remove this how are you getting the package out of the docker image after the docker image build?

josephbmanley replied:
Within Docker image

dejonghe commented on Jenkinsfile:
```diff
@@ -39,11 +40,6 @@ pipeline {
-
-        sh "python setup.py sdist"
- 
-        stash includes: "dist/semver-${env.SEMVER_NEW_VERSION}.tar.gz", name: 'PACKAGE' 
```
If you dont stash you may not have access to the package in the other stages.

josephbmanley replied:
Within Docker image

dejonghe commented on Jenkinsfile:
```diff
@@ -39,11 +40,6 @@ pipeline {
 
         // Docker build flags are set via the getDockerBuildFlags() shared library.
         sh "docker build ${getDockerBuildFlags()} -t rightbrainnetworks/auto-semver:${env.VERSION} ."
```
We can do this with native jenkins docker.build or something.

josephbmanley replied:
I'm unsure

dejonghe replied:
https://jenkins.io/doc/book/pipeline/docker/#building-containers

dejonghe reviewed (dismissed):
Everything looks good but the jenkins file doesnt look like it'll work, if it builds currently it may be because of prior builds having artifacts left around

yeslayla committed:
PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

PIPELINE: Use jenkin's docker library

yeslayla committed:
Remove line for beginning of `runAutoSemver.groovy`

yeslayla committed:
Add `getVersion.groovy` file

yeslayla committed:
Update README.md

dejonghe commented on Jenkinsfile:
```diff
@@ -38,12 +39,10 @@ pipeline {
-        sh "python setup.py sdist"
- 
-        stash includes: "dist/semver-${env.SEMVER_NEW_VERSION}.tar.gz", name: 'PACKAGE' 
+        script
```
you dont need script here

josephbmanley replied:
Required for `dockerImage` and `docker.build`

dejonghe commented on Jenkinsfile:
```diff
@@ -61,14 +60,20 @@ pipeline {
-              docker push rightbrainnetworks/auto-semver:${env.VERSION}
-              """)
+            sh("docker login -u ${DOCKER_USERNAME} -p ${DOCKER_PASSWORD}")
+            script
```
you dont need script here

josephbmanley replied:
Required for `dockerImage` object

dejonghe commented on Jenkinsfile:
```diff
@@ -61,14 +60,20 @@ pipeline {
-          sh "aws s3 cp `ls -t ./dist/semver-* | head -1` s3://rbn-ops-pkg-us-east-1/${env.SERVICE}/${env.SERVICE}-${env.VERSION}.tar.gz"
+          script
+          {
+            if("${env.SEMVER_STATUS}" == "0" && "${env.BRANCH_NAME}"  == "master")
```
better to be done with a when on a stage.

dejonghe replied:
you're doing that below..

yeslayla committed:
Merge master into 'hotfix/packageName'

yeslayla committed:
Move push latest & move to use PyPi token

yeslayla committed:
Install twine

josephbmanley commented:
As a note, I tried removing the some of the `script{}`s in the Jenkinsfile, but they were required to use methods attached to use the `docker.image` and `dockerImage` objects.

yeslayla committed:
Merge branch 'master' into hotfix/packageName

dejonghe reviewed (changes requested):
Trying to clear my open pr's, probably will never get implemented.
